packages/apps/dev-wallet/src/modules/network/network.service.ts

```diff
@@ -0,0 +1,58 @@
+import { getHostUrl, INetworkOptions } from '@kadena/client';
+import { INetwork } from './network.repository';
+
+export const fetchNetworkId = async (host: string) =>
+  fetch(host.endsWith('/') ? `${host}info` : `${host}/info`)
+    .then((response) => {
+      if (response.ok) {
+        return response.json();
+      }
+      throw new Error('Network is not healthy');
+    })
+    .then((data) => {
+      return data.nodeVersion;
+    })
+    .catch(() => {
+      return undefined;
+    });
+
+export const hostUrlGenerator = (networks: INetwork[]) => {
+  let healthyNetworks = networks;
+  let lastCheckTime = Date.now();
+  const checkNetworks = async () => {
+    lastCheckTime = Date.now();
+    healthyNetworks = await Promise.all(
+      networks.map(async (network) => {
+        const hosts = await Promise.all(
+          network.hosts.map(async (host) => {
+            const nodeVersion = await fetchNetworkId(host.url);
+            return {
+              ...host,
+              isHealthy: nodeVersion === network.networkId,
+              checkTime: Date.now(),
+            };
+          }),
+        );
+        return {
+          ...network,
+          hosts: hosts.filter((host) => host.isHealthy),
+        };
+      }),
+    );
+  };
+  checkNetworks();
+  return ({ networkId, chainId }: INetworkOptions) => {
+    if (Date.now() - lastCheckTime > 30000) {
+      checkNetworks();
```

Comment:
do we need to await `checkNetworks` at some time? Or return the promise

Reply:
not ready. it just checks the network health, I don't want to stop the whole process for this check
but I will add catch handler to it

Reply:
basically it filters the healthy networks for the next call not the current one